DevSkim-DotNet/Microsoft.DevSkim/ScopedRegexOperation.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.CST.OAT;
+using Microsoft.CST.OAT.Operations;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DevSkim
+{
+    public class ScopedRegexOperation : OatOperation
+    {
+        private RegexOperation regexEngine;
+        public ScopedRegexOperation(Analyzer analyzer) : base(Operation.Custom,analyzer)
+        {
+            CustomOperation = "ScopedRegex";
+            regexEngine = new RegexOperation(analyzer);
+            OperationDelegate = ScopedRegexOperationDelegate;
+            ValidationDelegate = ScopedRegexValidationDelegate;
+        }
+
+        public IEnumerable<Violation> ScopedRegexValidationDelegate(CST.OAT.Rule rule, Clause clause)
+        {
+            if (rule is null)
+            {
+                yield return new Violation($"Rule is null", new CST.OAT.Rule("RuleWasNull"));
+                yield break;
+            }
+            if (clause is null)
```

Comment:
Curious why each rule must have a clause since they are optional in the documentation same as today.